### PR TITLE
dts: arm: nrf54l: Add nodes required by MCUboot

### PR DESCRIPTION
**Pull request by michalek-no**
The commit adds nodes needed by MCUboot.
It also modifies chosen list by addition of MCUmgr port and code-partition.

### CODE REVIEW

**Review by nvlsianpu — COMMENTED** [review]
Only fixed-partitions are modified right now in this patch. Can you redact the commit message?

**Review by sigvartmh — APPROVED** [review]

**Review by nvlsianpu — DISMISSED** [review]

**Review by nvlsianpu — DISMISSED** [review]

File: boards/arm/nrf54l15pdk_nrf54l15/nrf54l15pdk_nrf54l15_cpuapp.dts
Comment (nordicjm) [inline boards/arm/nrf54l15pdk_nrf54l15/nrf54l15pdk_nrf54l15_cpuapp.dts:154]: Hate to be a pain but can we make this divisable by 4 so that external memory devices e.g. mx25 can be used, otherwise it is impossible? 348KiB for each of these then add 4KiB to settings to make that 36KiB

**Review by nordicjm — DISMISSED** [review]

**Review by nvlsianpu — DISMISSED** [review]

**Review by fabiobaltieri — DISMISSED** [review]

**Review by sigvartmh — APPROVED** [review]

**Comment by fabiobaltieri** [comment]
@anangl ping, do you folks want to merge this for 3.6? It's not quite a bugfix bug you may make an argument about it

**Review by anangl — DISMISSED** [review]

**Comment by fabiobaltieri** [comment]
> @anangl ping, do you folks want to merge this for 3.6? It's not quite a bugfix bug you may make an argument about it

@anangl ^^

**Comment by anangl** [comment]
I don't know if this is needed for 3.6. @michalek-no @nvlsianpu Is it?

**Comment by nvlsianpu** [comment]
@anangl We need this for NCS 2.6.0
Don't know whether this imply that it should go in v3.6

**Review by nvlsianpu — DISMISSED** [review]

File: Kconfig.zephyr
Comment (nvlsianpu) [reply Kconfig.zephyr:132]: might be moved to Kconfig.defconfig

**Review by nvlsianpu — DISMISSED** [review]

**Comment by michalek-no** [comment]
rebase

**Review by fabiobaltieri — DISMISSED** [review]

**Review by de-nordic — DISMISSED** [review]

**Review by nordicjm — APPROVED** [review]

File: boards/nordic/nrf54l15pdk/Kconfig.defconfig
Comment (nordicjm) [inline boards/nordic/nrf54l15pdk/Kconfig.defconfig:10]:
```suggestion
```

**Review by nordicjm — CHANGES_REQUESTED** [review]

**Review by nordicjm — APPROVED** [review]

**Review by fabiobaltieri — APPROVED** [review]

**Review by de-nordic — APPROVED** [review]

**Review by gchwier — APPROVED** [review]
Tested with PR https://github.com/zephyrproject-rtos/zephyr/pull/70105 (removed CONFIG_FPROTECT).
Building and flashing with enabled MCUboot works good on nrf54l15pdk,
used command:
`west build --sysbuild -p -b nrf54l15pdk/nrf54l15/cpuapp ${ZEPHYR_BASE}/samples/hello_world -d build-54l-hello -- \
-DSB_CONFIG_BOOTLOADER_MCUBOOT=y`

I was able also to run tests with `mcumgr`:
`${ZEPHYR_BASE}/scripts/twister -vv -T ${ZEPHYR_BASE}/tests/boot/with_mcumgr --west-flash --device-testing --device-serial /dev/ttyACM1 --force-platform -p nrf54l15pdk/nrf54l15/cpuapp --enable-slow -c`
Following test scenarios were successfully executed:
  _boot.with_mcumgr.test_upgrade.test_upgrade_with_confirm
  boot.with_mcumgr.test_upgrade.test_upgrade_with_revert
  boot.with_mcumgr.test_upgrade.test_upgrade_signature[no_key]
  boot.with_mcumgr.test_upgrade.test_upgrade_signature[invalid_key]
  boot.with_mcumgr.test_downgrade_prevention.test_downgrade_prevention_

**Comment by nvlsianpu** [comment]
@carlescufi Can this be merged?

**Comment by fabiobaltieri** [comment]
> @carlescufi Can this be merged?

Formally this needs @anangl approva, we have a dashboard and this does not even show as mergeable until the approver approved. Or we can change the approver, does not touch any core file so I suppose it's fine to move over to one of the mcubook collabs.

**Review by anangl — APPROVED** [review]

**Comment by fabiobaltieri** [comment]
That was easy, thanks @anangl :-)

**Pull request merged**
